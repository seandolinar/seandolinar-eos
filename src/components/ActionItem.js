import React from 'react';

import TemplateABI from './TemplateABI';


const ActionItem = ({data, obj, index, actionABI, methods}) => <li key={`${obj.account}-${index}`}>
    <div>{obj.account}</div>
    <div>{obj.name}</div>
    {!actionABI ?
        <button onClick={() => methods.getDataABI({ blockId: data.id, actionId: `${obj.account}-${index}`, name: obj.account })}>Get ABI</button> :
        <button onClick={() => methods.destroyDataABI({ blockId: data.id, actionId: `${obj.account}-${index}`, name: obj.account })}>Close ABI</button>
    }
    <TemplateABI actionABI={actionABI} actionData={obj} />
</li>


export default ActionItem;