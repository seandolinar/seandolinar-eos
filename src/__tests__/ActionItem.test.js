import React from 'react';
import renderer from 'react-test-renderer';

import ActionItem from '../components/ActionItem';

const mockProps = {
    obj: {account: 'testAccount', name: 'testName'},
    data: {id: 'a01', timestamp: '2019-09-13T13:01:04.500'},
    methods: {
        getDataABI: (mock) => mock,
        destroyDataABI: (mock) => mock
    },
    actionABI: {abi: {}},
    index: 3
}

it('matches previous snapshot: ActionItem', () => {
    expect(renderer.create(<ActionItem {...mockProps} />).toJSON()).toMatchSnapshot();
});
