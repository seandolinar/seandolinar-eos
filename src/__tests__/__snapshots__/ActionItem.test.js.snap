// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches previous snapshot: ActionItem 1`] = `
<li>
  <div>
    testAccount
  </div>
  <div>
    testName
  </div>
  <button
    onClick={[Function]}
  >
    Close ABI
  </button>
</li>
`;
